src/DeferredPromise.ts

```diff
@@ -0,0 +1,19 @@
+export interface DeferredPromise<T> extends Promise<T> {
+	resolve(value: T | PromiseLike<T>): void;
+	reject(reason?: any): void;
+}
+
+export function createDeferredPromise<T = void>(): DeferredPromise<T> {
+	let res: (value: T | PromiseLike<T>) => void;
+	let rej: (reason?: any) => void;
+
+	const promise = new Promise<T>((resolve, reject) => {
+		res = resolve;
+		rej = reject;
+	}) as DeferredPromise<T>;
+
+	promise.resolve = res!;
+	promise.reject = rej!;
+
+	return promise;
+}
```

Comment:
Am I missing something? (Sorry, reviewing on my phone)
I don't see this interface being used anywhere. Also I don't see any promise methods on the client.

Reply:
This is used here: https://github.com/ioBroker/socket-client/pull/5/files?file-filters%5B%5D=.ts#diff-f85526189b0a8634193b87cdbe47726022c55300e14a75004778ecca577983f2R162

What do you mean by promise methods on the client? The client is `EventEmitter`/socket.io compatible and we're building promisified wrappers around the emit-callback structure.

Reply:
Sorry, as I said: couldn't find it on the phone - found it on my PC ;-)